src/esp/sim/Simulator.cpp

```diff
@@ -664,6 +666,21 @@ void Simulator::removeObject(const int objectID,
       // resourceManager_->removeResourceByName(trajVisAssetName);
     }
   }
+  // if (deleteObjectNode) {
```

Comment:
Thanks everyone for the helpful feedback! @aclegg3 I did realize Simulator::removeObject was causing a crash, I think from a sensor being attached to a node that no longer exists. 

To fix this, I tried freeing the memory at the pointer to a sensor and/or erasing a sensor from a SensorSuite before PhysicsManager::removeObject is called. I also tried deleting the object scene node after the physical scene node and visual scene node were deleted, so that the destructor would delete things attached to the node. None of these worked, so I'm not sure what to do. 

Does anyone have any guidance for deleting a sensor when its node is deleted?

Reply:
One easy thing - any deleting here should be within `if (sceneHasPhysics(sceneID)) {` check.

Reply:
All the delete code should be within the `if (sceneHasPhysics(sceneID)) {` check.